src/ui/components/filters/facetscomponent.js

```diff
@@ -177,9 +185,9 @@ export default class FacetsComponent extends Component {
       this._filterbox.remove();
     }
 
-    let { filters } = this._state.get();
+    let { filters, resultsContext } = this._state.get();
 
-    if (!filters) {
+    if (!filters || resultsContext === ResultsContext.NO_RESULTS) {
```

Comment:
For now, would it just be easier to have the SearchDataTransformer._parseVerticalResponse return a response with an empty list of facets if ResultsContext.NO_RESULTS? Would that work?

Reply:
The future plan is to have some specific behavior for facets w/no results though right?

Reply:
I think bryan said they wanted something weird like when displaying no results, display all facets, and if any are applied reset the search to blank string and apply the facets

Reply:
We'd talked about how they might want this to work in the future. The current thinking is to have it behave like you said. That hasn’t been finalized in a spec though and a lot has been subject to change with noResults. I'd rather not leak the resultsContext implementation detail into Facets for now, if we can avoid it. It adds to the complexity of the component and serves a use-case we don't yet have final confirmation on. If we do get the go-ahead for the more complex behavior, we can always add resultsContext then and undo the small change to _parseVerticalResponse. We're not painting ourselves into a corner by leaving it out now and make the Facets component simpler by doing so. If you feel strongly, though, we can keep it in!

Reply:
Oh there was one more thing when I was testing this on friday, even if you have facets as [] the component container and things Facets title will still render. Otherwise I think I would definitely agree with you and just return an empty list of facets!

Reply:
Yeah, I didn't see this was in onMount. That could cause issues with the approach I was proposing too. The component might get mounted due to a valid query. If someone makes a subsequent query that returns noResults (and thus an empty array of facets), we'd need this component to disappear. But, I don't think onMount would get called again in that case, right? So, we definitely need to add this to setState as well.